### PR TITLE
Large scale fix of python-related files in torch/csrc/

### PR DESCRIPTION
**Pull request by goldsborough**

### CODE REVIEW

**Review by ezyang — APPROVED** [review]

File: .clang-tidy [hunk 1]
```diff
@@ -6,21 +6,24 @@ Checks: '
...
-  ,-cppcoreguidelines-interfaces-global-init
-  ,-cppcoreguidelines-owning-memory
-  ,hicpp-signed-bitwise
   ,hicpp-exception-baseclass
   ,hicpp-avoid-goto
   ,modernize-*
   ,-modernize-use-default-member-init
   ,-modernize-return-braced-init-list
   ,-modernize-use-auto
+  ,-modernize-use-using
```
Comment (goldsborough) [inline .clang-tidy]: Let me fixing this one

**Comment by pietern** [comment]
This will conflict with the `.clang-tidy` mods I did in #14491.

**Comment by ezyang** [comment]
@goldsborough I have some in flight diffs which would pass tidy after this change, but don't pass before. Should I NOLINT suppress, or just ship even with broken tidy?

**Comment by goldsborough** [comment]
@ezyang @pietern please NOLINT for now please (doesnt have to include the check name, I'll fix it after)

File: torch/csrc/jit/constants.cpp [hunk 2]
```diff
@@ -13,9 +13,10 @@ Value* insertConstant(
     c10::optional<ScopePtr> scope) {
   Node * n = g.create(prim::Constant);
   if(val.isTensor()) {
-    at::Tensor ref = std::move(val).toTensor();
+    at::Tensor ref = val.toTensor();
```
Comment (apaszke) [inline torch/csrc/jit/constants.cpp]: What was wrong with this `std::move`?

File: torch/csrc/utils/invalid_arguments.cpp [hunk 3]
```diff
@@ -124,25 +126,25 @@ std::vector<std::string> _splitString(const std::string &s, const std::string& d
...
     subtype.pop_back();
-    result.reset(new SequenceType(_buildType(subtype, false)));
+    result = torch::make_unique<SequenceType>(_buildType(subtype, false));
   } else {
-    result.reset(new SimpleType(type_name));
+    result = torch::make_unique<SimpleType>(type_name);
   }
   if (is_nullable)
-    result.reset(new NullableType(std::move(result)));
+    result = torch::make_unique<NullableType>(std::move(result));
```
Comment (apaszke) [inline torch/csrc/utils/invalid_arguments.cpp]: We could just replace those assignments with `return`

**Pull request closed without merging**